waspc/src/Wasp/Util/Debug.hs

```diff
@@ -0,0 +1,23 @@
+{-# OPTIONS_GHC -Wno-deprecations #-}
```

Comment:
OK I went a bit wild here, I added this "fake" pure function that checks if we are in debug mode.

I spent some time investigating other ways to accomplish running code conditionally depending on DEBUG env var, which included stuff like cabal flags and compiler pragmas, but all the solutions ended up boing overly complex, and this just seems in practice and at the moment to be the easiest. We can revisit it in the future but I would waste time trying to find something "cleaner". I did read online that doing unsafe lookup of env var like this shouldn't be able to cause any problems, so we should be ok.